packages/devextreme/js/__internal/events/core/m_events_engine.ts

```diff
@@ -1,685 +1,695 @@
+          getParents(parent);
         }
-    })),
+      };
+      getParents(element);
+      parents.push(window);
 
-    triggerHandler: getHandler(normalizeTriggerArguments(function(element, event, extraParameters) {
-        const handlersController = getHandlersController(element, event.type);
-        handlersController.callHandlers(event, extraParameters);
-    }))
+      let i = 0;
+
+      while (parents[i] && !event.isPropagationStopped()) {
+        const parentDataByEvent = getHandlersController(parents[i], event.type);
+        parentDataByEvent.callHandlers(extend(event, { currentTarget: parents[i] }), extraParameters);
+        i++;
+      }
+    }
+
+    if (element.nodeType || isWindow(element)) {
+      special.callMethod(eventName, '_default', element, [event, extraParameters]);
+      callNativeMethod(eventName, element);
+    }
+  })),
+
+  triggerHandler: getHandler(normalizeTriggerArguments((element, event, extraParameters) => {
+    const handlersController = getHandlersController(element, event.type);
+    handlersController.callHandlers(event, extraParameters);
+  })),
 });
 
 function applyForEach(args, method) {
-    const element = args[0];
+  const element = args[0];
 
-    if(!element) {
-        return;
-    }
+  if (!element) {
+    return;
+  }
 
-    if(domAdapter.isNode(element) || isWindow(element)) {
-        method.apply(eventsEngine, args);
-    } else if(!isString(element) && 'length' in element) {
-        const itemArgs = Array.prototype.slice.call(args, 0);
+  if (domAdapter.isNode(element) || isWindow(element)) {
+    method.apply(eventsEngine, args);
+  } else if (!isString(element) && 'length' in element) {
+    const itemArgs = Array.prototype.slice.call(args, 0);
 
-        Array.prototype.forEach.call(element, function(itemElement) {
-            itemArgs[0] = itemElement;
-            applyForEach(itemArgs, method);
-        });
-    } else {
-        throw errors.Error('E0025');
-    }
+    Array.prototype.forEach.call(element, (itemElement) => {
+      itemArgs[0] = itemElement;
+      applyForEach(itemArgs, method);
+    });
+  } else {
+    throw errors.Error('E0025');
+  }
 }
 
 function getHandler(method) {
-    return function() {
-        applyForEach(arguments, method);
-    };
+  return function () {
+    applyForEach(arguments, method);
+  };
 }
 
 function detectPassiveEventHandlersSupport() {
-    let isSupported = false;
-
-    try {
-        const options = Object.defineProperty({ }, 'passive', {
-            get: function() {
-                isSupported = true;
-                return true;
-            }
-        });
+  let isSupported = false;
+
+  try {
+    const options = Object.defineProperty({}, 'passive', {
+      get() {
+        isSupported = true;
+        return true;
+      },
+    });
 
-        window.addEventListener('test', null, options);
-    } catch(e) { }
+    (window as any).addEventListener('test', null, options);
+  } catch (e) {
+    //
```

Comment:
Should we really leave comment here? Maybe is there other ways to avoid warning?